app/src/parser_impl.c

```diff
@@ -316,8 +317,9 @@ parser_error_t  _appendCoinSymbol(char *value) {
     MEMZERO(buffer, sizeof(buffer));
     strcat(buffer, COIN_TICKER);
     strcat(buffer, " ");
-    strcat(buffer,value);
-    strcpy(value, buffer);
+    strcat(buffer, value);
+    // print length: strlen(value) + strlen(COIN_TICKER) + strlen(" ") + strlen("\0")
+    snprintf(value, strlen(value) + strlen(COIN_TICKER) + 2, "%s", buffer);
```

Comment:
Here we should know the maximum size for value and limit snprintf with that limit
maybe adding something like 
```
if (strlen(buffer) + 2 + strlen(COIN_TICKER) > outValLen-1) return error
```

Reply:
Ok, I did a refactor of this part. 
`_toStringCompactInt` now accepts a prefix value which for balances types will be `COIN_TICKER`.